### PR TITLE
fix #3504

### LINKED ISSUE #3504: Cargo doesnt exist
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

There is no cargo in the latest version of slimefun, the only error I have in the console is in the paste, I do not have more errors.: https://paste.walshy.dev/TT5xl96pZ6FV

### 📑 Reproduction Steps

----------

### 💡 Expected Behavior

Not working after update

### 📷 Screenshots / Videos

_No response_

### 📜 Server Log

https://paste.walshy.dev/TT5xl96pZ6FV

### 📂 `/error-reports/` folder

https://paste.walshy.dev/TT5xl96pZ6FV

### 💻 Server Software

Airplane

### 🎮 Minecraft Version

1.16.x

### ⭐ Slimefun version

Slimefun DEV - 1008 (git 020094e0) 

### 🧭 Other plugins

_No response_

### PR DESCRIPTION
**Pull request by J3fftw1**
## Description

This pr will change the way Slimefun handles colours materials
This will remove the ColoredMaterial class and will change completely in to tags
Right now i just have changed Cargo to support this.
When this proposal actually gets accepted i will change all other classes to also support tags

With the current implementation cargo actually already works this is fully tested

## Proposed changes

Remove ColoredMaterials class
Add different tags for colored materials 
Add a way to get a ordered list instead of a set
Change rainbowblocks, cargo and all others that currently use ColoredMaterial class to tags

## Related Issues (if applicable)

fix #3504

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/cargo/AbstractCargoNode.java [hunk 1]
```diff
@@ -92,6 +99,8 @@ protected void addChannelSelector(Block b, BlockMenu menu, int slotPrev, int slo
         boolean isChestTerminalInstalled = Slimefun.getIntegrations().isChestTerminalInstalled();
         int channel = getSelectedChannel(b);
 
+
+
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/cargo/AbstractCargoNode.java:103]:
```suggestion
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/ColoredMaterial.java [hunk 2]
```diff
@@ -199,29 +199,29 @@
...
+//            Material.GRAY_CANDLE,
+//            Material.LIGHT_GRAY_CANDLE,
+//            Material.CYAN_CANDLE,
+//            Material.PURPLE_CANDLE,
+//            Material.BLUE_CANDLE,
+//            Material.BROWN_CANDLE,
+//            Material.GREEN_CANDLE,
+//            Material.RED_CANDLE,
+//            Material.BLACK_CANDLE
+//    });
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/utils/ColoredMaterial.java:224]: why is this commented out? what's the alternative?

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/ColoredMaterial.java (hunk 2 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/utils/ColoredMaterial.java:224]: there isnt yet

This was for testing

Candles will be turned into tags aswell
Will do after i get feedback that something like this actually gets accepted

Comment (WalshyDev) [reply src/main/java/io/github/thebusybiscuit/slimefun4/utils/ColoredMaterial.java:224]: Don't see why we wouldn't. It is a breaking API change though.

Please cleanup your git history.

**Pull request closed without merging**